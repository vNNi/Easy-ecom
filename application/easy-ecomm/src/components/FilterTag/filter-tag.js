import { Container, Label } from './filter-tag.styles';

const FilterTag = ({ sort, category, onFilterClick }) => {
  return (
    <>
      <Container onClick={e => onFilterClick(e, { sort, category })}>
        <Label>{sort}</Label>
      </Container>
    </>
  );
};

export default FilterTag;
