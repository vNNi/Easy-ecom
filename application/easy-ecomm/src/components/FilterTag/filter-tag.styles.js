import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  height: fit-content;
  border-radius: 20px;
  background-color: ${({ theme }) => theme.colors.secondary};
  padding: 10px;
  cursor: pointer;
  :hover {
    padding: 12px;
  }
`;

export const Label = styled.span`
  text-transform: capitalize;
  color: ${({ theme }) => theme.text.light};
`;
