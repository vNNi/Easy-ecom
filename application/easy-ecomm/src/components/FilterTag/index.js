export { default } from './filter-tag';
