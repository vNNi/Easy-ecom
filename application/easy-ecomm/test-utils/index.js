import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import '@testing-library/jest-dom/extend-expect'; // jest-dom custom assertions
import 'jest-styled-components';

import Theme from '../src/styles/theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
