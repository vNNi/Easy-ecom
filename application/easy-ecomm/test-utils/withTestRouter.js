import React from 'react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';

export default function withTestRouter(tree, router = {}) {
  const {
    route = '',
    pathname = '',
    pagename = '',
    pagetype = '',
    query = {},
    asPath = '',
    push = async () => true,
    replace = async () => true,
    reload = () => null,
    back = () => null,
    prefetch = async () => undefined,
    beforePopState = () => null,
    isFallback = false,
    events = {
      emit: () => null,
      off: () => null,
      on: () => null,
    },
  } = router;

  return (
    <RouterContext.Provider
      value={{
        asPath,
        back,
        beforePopState,
        events,
        isFallback,
        pagename,
        pagetype,
        pathname,
        prefetch,
        push,
        query,
        reload,
        replace,
        route,
      }}
    >
      {tree}
    </RouterContext.Provider>
  );
}
