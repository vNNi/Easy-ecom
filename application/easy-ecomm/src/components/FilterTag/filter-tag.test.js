import FilterTag from './filter-tag';

import { renderWithTheme } from '../../../test-utils';

describe('[Components] - filter-tag', () => {
  test('should render correctly', () => {
    const el = renderWithTheme(<FilterTag />);
    expect(el).toMatchSnapshot();
  });
});
