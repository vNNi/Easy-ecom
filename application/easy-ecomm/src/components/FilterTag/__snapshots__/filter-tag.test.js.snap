// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Components] - filter-tag should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  border-radius: 20px;
  background-color: #fd9644;
  padding: 10px;
  cursor: pointer;
}

.c0:hover {
  padding: 12px;
}

.c1 {
  text-transform: capitalize;
  color: #ffffff;
}

<body>
    <div>
      <div
        class="c0"
      >
        <span
          class="c1"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter-tagstyles__Container-sc-1nuz32v-0 jATMDR"
    >
      <span
        class="filter-tagstyles__Label-sc-1nuz32v-1 ktKgsy"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
